packages/block-editor/src/components/rich-text/should-dismiss-pasted-files.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Given a collection of DataTransfer files and HTML and plain text strings,
+ * determine whether the files are to be dismissed in favor of the HTML.
+ *
+ * Certain office-type programs, like Microsoft Word or Apple Numbers,
+ * will, upon copy, generate a screenshot of the content being copied and
+ * attach it to the clipboard alongside the actual rich text that the user
+ * sought to copy. In those cases, we should let Gutenberg handle the rich text
+ * content and not the screenshot, since this allows Gutenberg to insert
+ * meaningful blocks, like paragraphs, lists or even tables.
+ *
+ * @param {File[]} files File objects obtained from a paste event
+ * @param {string} html  HTML content obtained from a paste event
+ * @return {boolean}     True if the files should be dismissed
+ */
+export function shouldDismissPastedFiles( files, html /*, plainText */ ) {
+	// The question is only relevant when there is actual HTML content and when
+	// there is exactly one image file.
+	if (
+		html &&
+		files?.length === 1 &&
+		files[ 0 ].type.indexOf( 'image/' ) === 0
+	) {
+		// An <img> tag found in the HTML source suggests that the content
+		// being pasted revolves around an image. Sometimes there are other
+		// elements found, like <figure>, but we assume that the user's
+		// intention is to paste the actual image file.
+		const IMAGE_TAG = /<\s*img\b/i;
+		return ! IMAGE_TAG.test( html );
```

Comment:
Should we make sure that there's only one image tag?

Reply:
Good question. Maybe.

Reply:
Pushed https://github.com/WordPress/gutenberg/pull/38992/commits/c83eeda60506bfae5a4d49d09c27f1f7fa37ef54

Reply:
Thinking more about it, it's actually possible we have multiple files and as a result multiple HTML tags. Maybe we can restrict it more by checking if the number of file is equal to the number of HTML tags found. Additionally, it may be good to check if the HTML contains any textual content?